assets/test/spec/services/apiServiceSpec.js

```diff
@@ -0,0 +1,294 @@
+         apiVersion: 'v1',
+         metadata: {},
+         spec: {}
+      }, {resource: 'routes', group: '', version: 'v1'}],
+      [{
+        apiVersion: 'v1',
+        kind: "DeploymentConfig",
+        metadata: {}
+      }, {resource: 'deploymentconfigs', group: '', version: 'v1'}]
+    ];
+
+    _.forEach(tc, _.spread(function(data, resource) {
+      it('derives a qualified resource form a provided data object', function() {
+          expect(APIService.deriveResource(data)).toEqual(resource);
+      });
+    }));
+  });
+
+  describe('#normalizeResource', function() {
+    // rather than using real examples like `build/log`, the strings here illustrate the desired behavior
+    describe('when the resource is all lowercase', function() {
+      it('provides the resource as is', function() {
+        var resource = 'alllowercase/beforeandafter';
+        expect(APIService.normalizeResource(resource)).toEqual(resource);
+      });
+    });
+
+    describe('when the resource contains mixed case before the slash', function() {
+      it('lower cases the first segment', function() {
+        var resource = 'MixedCaseStuff/beforeandafter';
+        expect(APIService.normalizeResource(resource)).toEqual('mixedcasestuff/beforeandafter');
+      });
+    });
+    describe('when the resource contains upper case after the slash', function() {
+      it('does nothing to post slash segment', function() {
+        var resource = 'FOO/BarBaz';
+        expect(APIService.normalizeResource(resource)).toEqual('foo/BarBaz');
+      });
+    });
+
+    describe('when the resource does not contain a slash', function() {
+      it('provides the resource as is', function() {
+        var resource = 'foobarbaz';
+        expect(APIService.normalizeResource(resource)).toEqual(resource);
+      });
+    });
+
+    describe('when the resource is falsy', function() {
+      var falsyValues = [null, undefined, ''];
+      _.each(falsyValues, function(isFalsy) {
+        it('provides undefined', function() {
+          expect(APIService.normalizeResource(isFalsy)).toEqual(undefined);
+        });
+      });
+    });
+
+  });
+
+  describe('#kindToResource', function() {
+    var tc = [
+      ['Test',                'tests'],
+      ['Route',               'routes'],
+      ['ImageStream',         'imagestreams'],
+      ['BuildConfig',         'buildconfigs'],
+      ['DeploymentConfig',    'deploymentconfigs'],
+      ['Service',             'services'],
+      ['Policy',              'policies'],
+      ['Identity',            'identities'],
+      ['ClusterRole',         'clusterroles'],
+      ['ComponentStatus',     'componentstatuses']
+    ];
+    _.forEach(tc, _.spread(function(kind, resource) {
+      // TODO: implies it actually knows, but really it just lowercase/pluralizes a string
+      // TEST: null, undefined, empty string... what should these do?
+      it('converts a kind to a resource by lowercasing & pluralizing the string', function() {
```

Comment:
Though not a fan of the style, I think you are right on uniqueness (though it runs fine) and this is probably the better way to loop these:

```javascript
it('does x', function() {
  _.each(tc, _.spread(function(a, b) {
     expect(func(a)).toEqual(b);
  }));
});
```
I think it works here since each item is proving the same behavior, its not a list of tests for different behaviors.